File: src/gui/src/profile/models/ProfileDto.ts
import type { Address } from "../../types/Address";
import type { Service } from "../../types/Service";
import type { TranslationLanguage } from "../../types/TranslationLanguage";

export default class ProfileDto {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string | null;
    address: Address | null;
    statusId?: number;
    sourceLanguages: TranslationLanguage[];
    targetLanguages: TranslationLanguage[];
    services: Service[];

    public constructor(
        id: string,
        firstName: string,
        lastName: string,
        email: string,
        phone: string | null = null,
        address: Address | null = null,
        statusId?: number,
        sourceLanguages: TranslationLanguage[] = [],
        targetLanguages: TranslationLanguage[] = [],
        services: Service[] = []
    ) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.statusId = statusId;
        this.sourceLanguages = sourceLanguages;
        this.targetLanguages = targetLanguages;
        this.services = services;
    }
}

File: src/backend/WordsmithHub.API/Features/Freelances/Models/ProfileDto.cs
using WordsmithHub.API.Features.Common.Models;
using WordsmithHub.Domain;

namespace WordsmithHub.API.Features.Freelances.Models;

public record ProfileDto
{
    public required Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public string? Phone { get; set; }
    public required string Email { get; set; }
    public required AddressDto? Address { get; set; }
    public required int StatusId { get; set; }
    public IReadOnlyList<TranslationLanguage> SourceLanguages { get; set; } = [];
    public IReadOnlyList<TranslationLanguage> TargetLanguages { get; set; } = [];
    public IReadOnlyList<Service> Services { get; set; } = [];
}
